### PR TITLE
chapter_13

### PR DESCRIPTION
**Pull request by starocketeer**

### CODE REVIEW

**Comment by starocketeer** [comment]
Ready

File: app/controllers/microposts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  end
+  
+  private
+  
+  def micropost_params
+      params.require(:micropost).permit :content, :image
+  end
+  
+  def correct_user
+      @micropost = current_user.microposts.find_by id: params[:id]
```
Comment (manhnt-0960) [inline app/controllers/microposts_controller.rb:34]: please check indent

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -10,6 +10,7 @@ def index
 
   def show
     @user = User.find_by params[:id]
+    @microposts = @user.microposts.paginate(page: params[:page])
```
Comment (manhnt-0960) [inline app/controllers/users_controller.rb:13]: add number of per page

File: app/models/micropost.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Micropost < ApplicationRecord
+  belongs_to :user
+  has_one_attached :image
+  scope -> { order created_at: :desc }
+  validates :user_id, presence: true
+  validates :content, presence: true, length: { maximum: Settings.validations.micropost.max_length }
+  validates :image, content_type: { in: %w[image/jpeg image/gif image/png],
+                                    message: "must be a valid image format" },
+                            size: { less_than: 5.megabytes,
```
Comment (manhnt-0960) [inline app/models/micropost.rb:9]: move 5 to settings.yml

File: app/controllers/microposts_controller.rb (hunk 1 above)
Comment (starocketeer) [reply app/controllers/microposts_controller.rb:34]: Can you please guide me? I don't understand

Comment (starocketeer) [reply app/controllers/microposts_controller.rb:34]: Done

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (starocketeer) [reply app/controllers/users_controller.rb:13]: I don't understand.

Comment (starocketeer) [reply app/controllers/users_controller.rb:13]: Done

File: app/models/micropost.rb (hunk 3 above)
Comment (starocketeer) [reply app/models/micropost.rb:9]: Done

**Comment by starocketeer** [comment]
Ready

**Comment by manhnt-0960** [comment]
merged

**Pull request merged**